api/python/t4/packages.py

```diff
@@ -378,20 +373,27 @@ def fetch(self, dest):
         """
         # TODO: do this with improved parallelism? connections etc. could be reused
         nice_dest = fix_url(dest).rstrip('/')
-        for key, entry in self._data.items():
-            entry.fetch('{}/{}'.format(nice_dest, quote(key)))
+        for logical_key, entry in self._walk():
+            entry.fetch('{}/{}'.format(nice_dest, quote(logical_key)))
 
     def keys(self):
         """
-        Returns list of logical_keys in the package.
+        Returns logical keys in the package.
         """
-        return list(self._data.keys())
+        return self._children.keys()
 
     def __iter__(self):
-        return iter(self._data)
+        return iter(self._children)
 
     def __len__(self):
-        return len(self._data)
+        return len(self._children)
+
+    def _walk(self, prefix=''):
```

Comment:
If we expose this as a public API method, does this also solve "[Implement a walk method on Package that traverses the package as a tree](https://app.asana.com/0/855836770921302/911647053675713)"?

Reply:
Asana is down for me, so can't see the task. What do we want the walk method to look like?

Reply:
(Asana is back up)

I tested this out in a notebook, and think this is exactly the intended `t4.Package.walk` behavior: a function that provides an iterator over the `(path, entry)` tuples.

So yeah, I suggest we expose this as a public method.